One way ANOVA

In [ ]:
# creating a dataset with 3 groups
mpg = c(34,35,34.3,35.5,35.8,35.3,36.5,36.4,37,37.6,33.3,34,34.7,33,34,33.9)
brand = factor(c("A","A","A","A","A","B","B","B","B","B","C","C","C","C","C","C"))

Lets see if there is any difference between average miles pre gallons across brands.

In [ ]:
mileage = data.frame(mpg = mpg, brand = brand)
mileage

mpg  brand
1  34.0 A    
2  35.0 A    
3  34.3 A    
4  35.5 A    
5  35.8 A    
6  35.3 B    
7  36.5 B    
8  36.4 B    
9  37.0 B    
10 37.6 B    
11 33.3 C    
12 34.0 C    
13 34.7 C    
14 33.0 C    
15 34.0 C    
16 33.9 C

Here:
- mpg is our dependent variable
- brand is our independent variable

In [ ]:
anova = aov(mpg~brand, data = mileage)
summary(anova)

            Df Sum Sq Mean Sq F value   Pr(>F)    
brand        2 20.596  10.298   19.05 0.000137 ***
Residuals   13  7.028   0.541                     
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

As we can see that our p-value is very low. We can say that our Anova result is significant implying that mean mpg differs accross all these brands.

# Two way ANOVA

In [3]:
data = read.csv("/content/tw.txt")
head(data, 5)

Format   Subject    Satisfaction
1 Online   English    5           
2 Hybrid   Statistics 8           
3 In-Class Science    7           
4 Online   Math       4           
5 Hybrid   English    6

Here, satisfaction is the dependent variable.

'Format:Subject' means the interaction between two independent variables.

In [6]:
res = aov(Satisfaction ~ Format + Subject + Format:Subject, data = data)
summary(res)

               Df Sum Sq Mean Sq F value   Pr(>F)    
Format          2 173.69   86.85  453.12  < 2e-16 ***
Subject         3  27.60    9.20   47.99 6.03e-16 ***
Format:Subject  6  12.86    2.14   11.18 2.43e-08 ***
Residuals      60  11.50    0.19                     
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

# Chi-Square test of goodness of fit

In [9]:
# creating a dataset with observations and expectations
obs = c(195, 165, 47, 15, 30, 35, 8, 5)
pct = c(0.374, 0.357, 0.085, 0.034, 0.066, 0.063, 0.015, 0.006)

This dataset contains bloodtype, population prevalance rate, count of the bloodtype and the expected bloodtype based on populatiuon prevalance

In [10]:
my_data = data.frame(Blood_type = c("O+", "A+", "B+", "AB+", "O-", "A-",
"B-", "AB-"), pct, obs, Exp = sum(obs)*pct)

In [11]:
head(my_data, 5)

Blood_type pct   obs Exp  
1 O+         0.374 195 187.0
2 A+         0.357 165 178.5
3 B+         0.085  47  42.5
4 AB+        0.034  15  17.0
5 O-         0.066  30  33.0

In [13]:
# x stands for observed value
# p stands for expected probabilities
chisq.test(x = my_data$obs , p = my_data$pct)

Warning message in chisq.test(x = my_data$obs, p = my_data$pct):
“Chi-squared approximation may be incorrect”



	Chi-squared test for given probabilities

data:  my_data$obs
X-squared = 4.1033, df = 7, p-value = 0.7678


As our p-value is very high i.e., 0.7678, we *cannot* reject our null hypothesis.

# Chi-Square test of factor independece

In [14]:
monthly = c(91, 150, 109)
occasionally = c(90, 200, 198)
never = c(51, 155,172)

In [15]:
df = data.frame(monthly, occasionally, never)

In [17]:
row.names(df) = c("under 45", "45-59", "60 and over")

In [18]:
df

monthly occasionally never
under 45     91      90           51  
45-59       150     200          155  
60 and over 109     198          172

This dataset shows distribution of people based on age and how frequently they consume alcohol.

The age and the frequency at which someone drinks are related or not?

In [19]:
chisq.test(df)


	Pearson's Chi-squared test

data:  df
X-squared = 25.086, df = 4, p-value = 4.835e-05


As p-value is very low, we can say that there is dependence between factor age and the frequency of which someone drinks.

